### PR TITLE
Replace shadowed/shadowing variables

### PR DESCRIPTION
**Pull request by dbenage-cx**
Majority are renames, a few cases were dropped where they matched the previous definition.

The last commit attempts to resolve an issue with `GraphicalSummary::parent`, while the resolution works on my machine, I'm cautious it is not correct.

Only remaining noted warnings for `-Wshadow` are for utf8/checked, I left those for a potential update of library.

### CODE REVIEW

File: UI/CombatReport/GraphicalSummary.cpp [hunk 1]
```diff
@@ -655,7 +655,7 @@ class OptionsBar : public GG::Wnd {
             button(nullptr)
         {
             button = Wnd::Create<CUIButton>("-");
-            parent->AttachChild(button);
+            parent_->AttachChild(button);
```
Comment (geoffthemedio) [inline UI/CombatReport/GraphicalSummary.cpp:658]: @LGM-Doyle Should this be happening in `CompleteConstruction` ?

File: GG/src/Font.cpp [hunk 2]
```diff
@@ -1580,13 +1580,13 @@ void Font::FillTemplatedText(
         std::shared_ptr<TextElement> elem = (*te_it);
 
         // For each character in the TextElement.
-        auto it = elem->text.begin();
+        auto text_it = elem->text.begin();
```
Comment (geoffthemedio) [inline GG/src/Font.cpp:1583]: is this change necessary? not that it's a problem, I just don't see where else in this scope that `it` is used...

File: GG/src/Menu.cpp [hunk 3]
```diff
@@ -45,10 +45,10 @@ MenuItem::MenuItem() :
     MenuItem("", false, false)
 {}
 
-MenuItem::MenuItem(bool separator) :
+MenuItem::MenuItem(bool separator_) :
     disabled(true),
     checked(false),
-    separator(true),
+    separator(separator_),
```
Comment (geoffthemedio) [inline GG/src/Menu.cpp]: This is changing behaviour. The parameter is currently ignored.

File: UI/DesignWnd.cpp [hunk 4]
```diff
@@ -4339,19 +4339,19 @@ void DesignWnd::MainPanel::ReplaceDesign() {
     m_type_to_create = old_m_type_to_create;
 
     // If replacing a saved design
-    if (const auto replaced_design = EditingSavedDesign()) {
+    if (const auto saved_maybe_design = EditingSavedDesign()) {
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp]: seems unnecessary...

File: UI/ObjectListWnd.cpp [hunk 5]
```diff
@@ -2567,8 +2567,8 @@ void ObjectListWnd::ObjectRightClicked(GG::ListBox::iterator it, const GG::Pt& p
             auto produce_building_action = [this, entry, app, cur_empire, &focus_ship_building_common_action]() {
                 std::string bld = entry.first;
                 bool needs_queue_update(false);
-                for (const auto& entry : m_list_box->Selections()) {
-                    ObjectRow *row = dynamic_cast<ObjectRow *>(entry->get());
+                for (const auto& selection_it : m_list_box->Selections()) {
```
Comment (geoffthemedio) [inline UI/ObjectListWnd.cpp]: The type of `Selections` is a container of `iterator`, but in this case the variable `entry` -> `selection_it` isn't an iterator over the container `Selections`, so I wouldn't name it with `_it` in the name.

File: UI/OptionsWnd.cpp [hunk 6]
```diff
@@ -925,7 +925,7 @@ GG::Spin<int>* OptionsWnd::IntOption(GG::ListBox* page, int indentation_level, c
     text_control->SetBrowseModeTime(GetOptionsDB().Get<int>("UI.tooltip-delay"));
     text_control->SetBrowseText(UserString(GetOptionsDB().GetDescription(option_name)));
     spin->ValueChangedSignal.connect(
-        [option_name](const int& value){ GetOptionsDB().Set(option_name, value); });
+        [option_name](const int& option_value){ GetOptionsDB().Set(option_name, option_value); });
```
Comment (geoffthemedio) [inline UI/OptionsWnd.cpp]: `new_value` perhaps?

File: UI/ProductionWnd.cpp [hunk 7]
```diff
@@ -135,15 +135,15 @@ namespace {
             else
                 myQuantSet.insert(quantity);
 
-            for (int quantity : myQuantSet) {
-                auto row =  GG::Wnd::Create<QuantRow>(quantity, build.item.design_id, nwidth, h, inProgress, amBlockType);
+            for (int next_quantity : myQuantSet) {
```
Comment (geoffthemedio) [inline UI/ProductionWnd.cpp]: maybe `droplist_quantity` ?

File: UI/SidePanel.cpp [hunk 8]
```diff
@@ -3033,8 +3033,8 @@ void SidePanel::RefreshInPreRender() {
 }
 
 void SidePanel::RefreshSystemNames() {
-    auto system = GetSystem(s_system_id);
-    if (!system)
+    auto this_system = GetSystem(s_system_id);
```
Comment (geoffthemedio) [inline UI/SidePanel.cpp]: `selected_system`

Comment (geoffthemedio) [reply UI/SidePanel.cpp]: although checking the selected system can probably just be commented out entirely, as the result doesn't seem to be used anywhere later.

File: UI/Sound.cpp [hunk 9]
```diff
@@ -392,7 +392,7 @@ void Sound::Impl::PlayMusic(const boost::filesystem::path& path, int loops /* =
     ALenum m_openal_error;
     std::string filename = PathToString(path);
     FILE* m_f = nullptr;
-    vorbis_info* vorbis_info;
+    vorbis_info* vorbis_info_ptr;
```
Comment (geoffthemedio) [inline UI/Sound.cpp:395]: why?

File: universe/Universe.cpp [hunk 10]
```diff
@@ -238,8 +238,8 @@ std::set<int> Universe::EmpireVisibleObjectIDs(int empire_id/* = ALL_EMPIRES*/)
     // if an empire has visibility of it
     for (auto obj_it = m_objects.const_begin(); obj_it != m_objects.const_end(); ++obj_it) {
         int id = obj_it->ID();
-        for (int empire_id : empire_ids) {
-            Visibility vis = GetObjectVisibilityByEmpire(id, empire_id);
+        for (int next_empire_id : empire_ids) {
```
Comment (geoffthemedio) [inline universe/Universe.cpp]: `detector_empire_id`

File: universe/Universe.cpp [hunk 11]
```diff
@@ -2726,8 +2721,8 @@ std::set<int> Universe::RecursiveDestroy(int object_id) {
         Destroy(object_id);
         retval.insert(object_id);
 
-    } else if (auto fleet = std::dynamic_pointer_cast<Fleet>(obj)) {
-        for (int ship_id : fleet->ShipIDs()) {
+    } else if (auto obj_fleet = std::dynamic_pointer_cast<Fleet>(obj)) {
```
Comment (geoffthemedio) [inline universe/Universe.cpp:2724]: why?

File: universe/Universe.cpp [hunk 12]
```diff
@@ -2738,8 +2733,8 @@ std::set<int> Universe::RecursiveDestroy(int object_id) {
         Destroy(object_id);
         retval.insert(object_id);
 
-    } else if (auto planet = std::dynamic_pointer_cast<Planet>(obj)) {
-        for (int building_id : planet->BuildingIDs()) {
+    } else if (auto obj_planet = std::dynamic_pointer_cast<Planet>(obj)) {
```
Comment (geoffthemedio) [inline universe/Universe.cpp:2736]: why?

File: universe/ValueRef.cpp [hunk 13]
```diff
@@ -97,11 +97,11 @@ namespace {
 
     // Generates a debug  trace that can be included in error logs, augmenting the ReconstructName() info with
     // additional info identifying the object references that were successfully followed.
-    std::string TraceReference(const std::vector<std::string>& property_name, ValueRef::ReferenceType ref_type,
+    std::string TraceReference(const std::vector<std::string>& property_names, ValueRef::ReferenceType ref_type,
```
Comment (geoffthemedio) [inline universe/ValueRef.cpp]: As far as I recall, it is just a single property name, not multiple property names. The vector contains the parts of the name broken up, eg. `Source.Planet.Population` -> `{Source, Planet, Population}`

File: universe/ValueRef.cpp [hunk 14]
```diff
@@ -932,8 +932,8 @@ int Variable<int>::Eval(const ScriptingContext& context) const
 
     }
     else if (property_name == "LastTurnBattleHere") {
-        if (auto system = std::dynamic_pointer_cast<const System>(object))
-            return system->LastTurnBattleHere();
+        if (auto const_system = std::dynamic_pointer_cast<const System>(object))
```
Comment (geoffthemedio) [inline universe/ValueRef.cpp:935]: seems unnecessary, but OK...

**Comment by geoffthemedio** [comment]
Otherwise looks OK, but haven't tested.

File: GG/src/Font.cpp (hunk 2 above)
Comment (dbenage-cx) [reply GG/src/Font.cpp:1583]: Chose to replace the shadowed var, the shadowing case is on line 1591 `auto it = m_glyphs.find(c);`

File: GG/src/Menu.cpp (hunk 3 above)
Comment (dbenage-cx) [reply GG/src/Menu.cpp]: Dropped this change

File: UI/DesignWnd.cpp (hunk 4 above)
Comment (dbenage-cx) [reply UI/DesignWnd.cpp]: Reverted

File: UI/ObjectListWnd.cpp (hunk 5 above)
Comment (dbenage-cx) [reply UI/ObjectListWnd.cpp]: Changed to `selection` (consistent with previous for loop)

File: UI/Sound.cpp (hunk 9 above)
Comment (dbenage-cx) [reply UI/Sound.cpp:395]: gcc complains with `shadows a global declaration` from [vorbis/codec.h](https://github.com/xiph/vorbis/blob/master/include/vorbis/codec.h#L53)
```

File: universe/Universe.cpp (hunk 11 above)
Comment (dbenage-cx) [reply universe/Universe.cpp:2724]: warning for shadowed on line 2762 `for (auto& fleet : m_objects.FindObjects<Fleet>())`

File: universe/Universe.cpp (hunk 12 above)
Comment (dbenage-cx) [reply universe/Universe.cpp:2736]: warning for shadowed on line 2776 `auto planet = GetPlanet(building->PlanetID());`

File: universe/ValueRef.cpp (hunk 13 above)
Comment (dbenage-cx) [reply universe/ValueRef.cpp]: Changed to renaming shadowing var to `property_name_part`, consistent with `ReconstructNames`

File: universe/ValueRef.cpp (hunk 14 above)
Comment (dbenage-cx) [reply universe/ValueRef.cpp:935]: complains of shadow on line 937 `else if (auto system = GetSystem(object->SystemID()))`

**Comment by dbenage-cx** [comment]
Ty for review and suggestions for names.

File: universe/Universe.cpp (hunk 12 above)
Comment (geoffthemedio) [reply universe/Universe.cpp:2736]: The scopes are separate, I think...  But would probably be better to rename the other one, for consistency in naming with the other if/else test variable names.

File: UI/CombatReport/GraphicalSummary.cpp (hunk 1 above)
Comment (LGM-Doyle) [reply UI/CombatReport/GraphicalSummary.cpp:658]: This looks correct as is.

`ToggleData` is not  derived from `GG::Wnd`.  When `ToggleData` is created in `OptionsBar::CompleteConstruction()` both parent `OptionsBar` has a well defined `shared_from_this()`.

File: universe/Universe.cpp (hunk 12 above)
Comment (dbenage-cx) [reply universe/Universe.cpp:2736]: I was not aware until this PR:  this variables scope (from assignment in conditional) lasts until the end of the full block, not just within the success portion.
Clearest example I've come across is http://en.cppreference.com/w/cpp/language/if (bottom of page), `x` is still in scope within the else
Could break to separate if blocks with early returns

Comment (geoffthemedio) [reply universe/Universe.cpp:2736]: huh... good to know.
